src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -288,20 +300,262 @@ private static ProjectItemGroupElement CreateItemGroup(Project project, string f
+        /// <param name="assetsFile">Assets file for all targets and libraries</param>
+        /// <param name="transitive">Include transitive packages in the result</param>
+        /// <returns></returns>
+        internal IEnumerable<FrameworkPackages> GetResolvedVersions(
+            string projectPath, IEnumerable<string> userInputFrameworks, LockFile assetsFile, bool transitive)
+        {
+            if (userInputFrameworks == null)
+            {
+                throw new ArgumentNullException(nameof(userInputFrameworks));
+            }
+
+            if (projectPath == null)
+            {
+                throw new ArgumentNullException(nameof(projectPath));
+            }
+
+            if (assetsFile == null)
+            {
+                throw new ArgumentNullException(nameof(assetsFile));
+            }
+
+            var resultPackages = new List<FrameworkPackages>();
+            var requestedTargetFrameworks = assetsFile.PackageSpec.TargetFrameworks;
+            var requestedTargets = assetsFile.Targets;
+
+            // If the user has entered frameworks, we want to filter
+            // the targets and frameworks from the assets file
+            if (userInputFrameworks.Any())
+            {
+                //Target frameworks filtering
+                var parsedUserFrameworks = userInputFrameworks.Select(f =>
+                                               NuGetFramework.Parse(f.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Select(s => s.Trim()).ToArray()[0]));
+                requestedTargetFrameworks = requestedTargetFrameworks.Where(tfm => parsedUserFrameworks.Contains(tfm.FrameworkName)).ToList();
+
+                //Assets file targets filtering by framework and RID
+                var filteredTargets = new List<LockFileTarget>();
+                foreach (var frameworkAndRID in userInputFrameworks)
+                {
+                    var splitFrameworkAndRID = frameworkAndRID.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Select(s => s.Trim()).ToArray();
+                    // If a / is not present in the string, we get all of the targets that
+                    // have matching framework regardless of RID.
+                    if (splitFrameworkAndRID.Count() == 1)
+                    {
+                        filteredTargets.AddRange(requestedTargets.Where(target => target.TargetFramework.Equals(NuGetFramework.Parse(splitFrameworkAndRID[0]))));
+                    }
+                    else
+                    {
+                        //RID is present in the user input, so we filter using it as well
+                        filteredTargets.AddRange(requestedTargets.Where(target => target.TargetFramework.Equals(NuGetFramework.Parse(splitFrameworkAndRID[0])) &&
+                                                                                  target.RuntimeIdentifier != null && target.RuntimeIdentifier.Equals(splitFrameworkAndRID[1])));
```

Comment:
`target.RuntimeIdentifier.Equals(splitFrameworkAndRID[1])` 

this should be case insensitive comparison